src/Driver/Http1Driver.php

```diff
@@ -475,25 +479,25 @@ private function parser(): \Generator
                     );
                 }
 
-                if (isset($headers["expect"][0]) && \strtolower($headers["expect"][0]) === "100-continue") {
+                if (isset($headerMap["expect"][0]) && \strtolower($headerMap["expect"][0]) === "100-continue") {
                     yield $this->write(
-                        new Request($this->client, $method, $uri, $headers, null, $protocol),
+                        new Request($this->client, $method, $uri, $headerMap, null, $protocol),
                         new Response(Status::CONTINUE, [])
                     );
                 }
 
                 // Handle HTTP/2 upgrade request.
                 if ($protocol === "1.1"
-                    && isset($headers["upgrade"][0], $headers["http2-settings"][0], $headers["connection"][0])
+                    && isset($headerMap["upgrade"][0], $headerMap["http2-settings"][0], $headerMap["connection"][0])
                     && !$this->client->isEncrypted()
                     && $this->options->isHttp2UpgradeAllowed()
-                    && false !== \stripos($headers["connection"][0], "upgrade")
-                    && \strtolower($headers["upgrade"][0]) === "h2c"
-                    && false !== $h2cSettings = \base64_decode(\strtr($headers["http2-settings"][0], "-_", "+/"), true)
+                    && false !== \stripos($headerMap["connection"][0], "upgrade")
+                    && \strtolower($headerMap["upgrade"][0]) === "h2c"
+                    && false !== $h2cSettings = \base64_decode(\strtr($headerMap["http2-settings"][0], "-_", "+/"), true)
                 ) {
                     // Request instance will be overwritten below. This is for sending the switching protocols response.
                     yield $this->write(
-                        new Request($this->client, $method, $uri, $headers, null, $protocol),
+                        new Request($this->client, $method, $uri, $headerMap, null, $protocol),
```

Comment:
Shouldn't this also use the `$headers` variables and add the headers manually?

Reply:
Same as above.